real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/Map.java

```diff
@@ -2,89 +2,257 @@
+                // TODO Auto-generated method stub
+
+            }
+        });
+        getMapTree().getViewer().addMouseMotionListener(new MouseMotionListener() {
+            @Override
+            public void mouseMoved(MouseEvent e) {
+                // TODO Auto-generated method stub
+
+            }
+
+            @Override
+            public void mouseDragged(MouseEvent e) {
+                // TODO Auto-generated method stub
+
+                zoomLevel = getMapTree().getViewer().getZoom();
+
+                mapViewerLat -= ((mapDragY - e.getY()) * 0.001) / (zoomLevel * 1000);
+                mapViewerLon -= ((e.getX() - mapDragX) * 0.001) / (zoomLevel * 1000);
+                getMapTree().getViewer().setDisplayPosition(new Coordinate(mapViewerLat, mapViewerLon), zoomLevel);
+            }
+        });
+    }
+
+    private void addCacheToTree() {
+        try {
+            TileCache courseCache = new MemoryTileCache();
+            File mapCacheDir = new File(mapCacheFolderDiskPath);
+            if(!mapCacheDir.isDirectory() || !mapCacheDir.exists()) {
+                throw new IOException("cache dir isn't properly structured or doesn't exist");
+            }
+
+            FilenameFilter filter = (dir, name) -> {
+                // TODO Auto-generated method stub
+                if(name.contains("png")) {
+                    return true;
+                }
+                return false;
+            };
+            String[] cachedImages = mapCacheDir.list(filter);
+            if (cachedImages == null) {
+                return;
+            }
+            for(String imageName : cachedImages) {
+                BufferedImage tileImageSource = ImageIO.read(new File(mapCacheDir.getAbsolutePath() + "/" + imageName));
+                String[] tileCoords = imageName.substring(0, imageName.indexOf(".")).split("_");
+                int xCoord = Integer.parseInt(tileCoords[0]);
+                int yCoord = Integer.parseInt(tileCoords[1]);
+                zoomLevel = Integer.parseInt(tileCoords[2]);
+
+                Tile cacheInsert = new Tile(getMapTree().getViewer().getTileController().getTileSource(),
+                        xCoord, yCoord, zoomLevel, tileImageSource);
+                cacheInsert.setLoaded(true);
+                courseCache.addTile(cacheInsert);
+            }
+
+            getMapTree().getViewer().getTileController().setTileCache(courseCache);
+        }
+        catch (IOException e) {
+            new RobobuggyLogicNotification("Something is wrong with the map cache: " + e.getMessage(), RobobuggyMessageLevel.EXCEPTION);
+        }
+    }
+
+    /**
+     * updates the current arrow displaying on the GUI - shows orientation based on GPS
+     */
+    public void updateArrow() {
+		List<MapPolygon> polygons = getMapTree().getViewer().getMapPolygonList();
+			List<MapMarker> markers = getMapTree().getViewer().getMapMarkerList();
+			if (markers.size() >= 2) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2362883633/issues/source?bid=2938665&fileBranchId=3216826#l178)